orttraining/orttraining/python/training/execution_agent.py

```diff
@@ -107,33 +109,33 @@ def __init__(self, path_or_bytes, session_options=None, providers=None, provider
         self._training_agent = None
         self._inference_session = None
 
-        self.create_training_agent(path_or_bytes, session_options, providers, provider_options)
+        self.create_training_agent(
+            path_or_bytes, session_options, providers, provider_options, fw_feed_names, fw_fetches_names,
+            fw_outputs_device_info, bw_feed_names, bw_fetches_names, bw_outputs_device_info)
 
-    def create_training_agent(self, path_or_bytes, session_options, providers, provider_options):
+    def create_training_agent(self, path_or_bytes, session_options, providers, provider_options,
+                              fw_feed_names, fw_fetches_names, fw_outputs_device_info, bw_feed_names, bw_fetches_names,
+                              bw_outputs_device_info):
         self._inference_session = onnxruntime.InferenceSession(path_or_bytes, session_options,
                                                                providers, provider_options)
-        self._training_agent = C_TrainingAgent(self._inference_session._sess)
+        self._training_agent = C_TrainingAgent(self._inference_session._sess, fw_feed_names, fw_fetches_names,
+                                               fw_outputs_device_info, bw_feed_names, bw_fetches_names, bw_outputs_device_info)
```

Comment:
Passing and defining all those arguments makes this code harder to read, and the `create_training_agent()` method name doesn't add much:

```suggestion
        self._inference_session = onnxruntime.InferenceSession(path_or_bytes, session_options,
                                                               providers, provider_options)
        self._training_agent = C_TrainingAgent(self._inference_session._sess, fw_feed_names, fw_fetches_names,
                                               fw_outputs_device_info, bw_feed_names, bw_fetches_names, bw_outputs_device_info)
```